content/articles/how-sweet-is-kubernetes/index.md

```diff
@@ -0,0 +1,273 @@
+---
+layout: engineering-education
+status: publish
+published:true 
+url: /engineering-education/how-sweet-is-kubernetes/
+title: How sweet is Kubernetes?
+description: This article simplifies concepts of Kubernetes. It starts with demystifying the basic concept of micro-service using Fleetman Tracking Application which is built on a simple micro-service architecture, using it further to explain the concepts of Kubernetes.
+author: qoyum-yusuf
+date: 2021-05-01T00:00:00-08:00
+topics: [HOW SWEET IS KUBERNETES?]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/quoyum-yusuf/hero.jpeg
+    alt: How sweet is Kubernetes cover image 
+---
+
+### Introduction
+What was your first impression when you read Kubernetes?
+I bet you were already thinking, "Oh no, here comes another tech buzzword. Already sounds like a difficult concept."
+Well, you are not so right. However, let us dive in!
+We will be using the Fleetman application to explain the concept of Kubernetes.
+
+**Note: You can clone the GitHub repository used to describe this tutorial [here](https://github.com/wizardcalidad/deploying_fleetman_application_with_kubernetes).**
+Let us get staarted.
+
+### Overview of the Fleetman application.
+The Fleetman Application is an application that tracks the location of trucks as they move around the city of London. Those trucks are each equipped with the Global Positioning System(GPS) and periodically report their position to the central server. It is a simple microservice architecture with little to no complexity. 
+This article is not about the Fleetman application. Instead, we will be using Fleetman to understand our content better.
+Microservices obey the **SOLID** principle, which is:
+- `Single Responsibility Principle`, each part of the architecture should do just one thing.
+- `The Position Simulation` only simulates the positions of the truck at any point in time, and that is its only job.
+Once it reads the position, it hands over the data to the `Message Queue`. The essence of a queue in any architecture of microservice is to avoid loss of data. The queue receives the data, keeping it waiting for the request from the position tracker. The queue implemented here is **ActiveM.** It is also utilized as a docker container. The position tracker gets the information from the queue and performs many calculations, like calculating the speed a particular truck has covered. **API Gateway** is the single entry point to the entire application. Its work is to match all requests incoming to underlying microservices.
+![Fleetman microservice](/engineering-education/qoyum_yusuf/fleetman.png)
```

Comment:
I forgot to mention this. This is how we link images.
So it's normally engineering education - > forked branch name -> name of the image file. 
So that's how you will be linking all your images.

Reply:
Okay thanks a bunch,
This is for later posts right?

Reply:
Yes, it is.